### PR TITLE
Enlist reactive entity for reflection

### PR DESCRIPTION
**Pull request by loicmathieu**
MongoDB with panache enlist for reflection entities (and repository parameterized types) automatically.

This PR adds the same facility for the reactive implementation.

This has been discuss on Zuilip: https://quarkusio.zulipchat.com/#narrow/stream/187030-users/topic/Bson.20codec.20error.20when.20using.20reactive.20mongo.20panache.20in.20native

### CODE REVIEW

File: extensions/panache/mongodb-panache/deployment/src/main/java/io/quarkus/mongodb/panache/deployment/PanacheResourceProcessor.java [hunk 1]
```diff
@@ -336,8 +347,13 @@ void buildMutiny(CombinedIndexBuildItem index,
             if (modelClasses.add(classInfo.name().toString()))
                 modelEnhancer.collectFields(classInfo);
         }
+
+        // iterate over all the entity classes
```
Comment (machi1990) [inline extensions/panache/mongodb-panache/deployment/src/main/java/io/quarkus/mongodb/panache/deployment/PanacheResourceProcessor.java]: do we need these comments?

Comment (loicmathieu) [reply extensions/panache/mongodb-panache/deployment/src/main/java/io/quarkus/mongodb/panache/deployment/PanacheResourceProcessor.java]: I copied them for the implementation of the impertive version.
this method is really big and complex so the comments help me to maintain it yes ...

Comment (machi1990) [reply extensions/panache/mongodb-panache/deployment/src/main/java/io/quarkus/mongodb/panache/deployment/PanacheResourceProcessor.java]: I personally do not see if they add any value.. upon arriving here the loop tells me that an iteration is going on..same for the one below about reflection registration.

Comment (FroMage) [reply extensions/panache/mongodb-panache/deployment/src/main/java/io/quarkus/mongodb/panache/deployment/PanacheResourceProcessor.java]: While I also don't think this particular comment adds much value, I think it's way overkill to ask people to put less comments. In general comments are good and should be encouraged.

**Review by FroMage — APPROVED** [review]
LGTM.

**Pull request merged**